### PR TITLE
Improve extensibility in Applications

### PR DESCRIPTION
**Pull request by induwaram**
### Purpose
Improving extensibility in Applications by adding an overridden tab feature, Customized tab feature, customized label feature.

### CODE REVIEW

File: apps/console/src/features/applications/components/danger-zone.tsx [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+   interface DangerZonePropsInterface extends SBACInterface<FeatureConfigInterface>, IdentifiableComponentInterface {
+    /**
+     * Currently editing application id.
+     */
+    appId?: string;
+    /**
+     * Name of the application.
+     */
+    name: string;
+   }
```
[the remark this answers is not in the document]
Comment (induwaram) [reply apps/console/src/features/applications/components/danger-zone.tsx]: Fixed with https://github.com/wso2/identity-apps/pull/2983/commits/a582cef64780cf4b9ffe922d0a8adda99e94f418

File: apps/console/src/features/applications/components/danger-zone.tsx [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+    const UIConfig: UIConfigInterface = useSelector((state: AppState) => state?.config?.ui);
+
+    const [ showDeleteConfirmationModal, setShowDeleteConfirmationModal ] = useState<boolean>(false);
+    const [ isDeletionInProgress, setIsDeletionInProgress ] = useState<boolean>(false);
+
+
+    const { t } = useTranslation();
+    const dispatch = useDispatch();
+
+
```
[the remark this answers is not in the document]
Comment (induwaram) [reply apps/console/src/features/applications/components/danger-zone.tsx]: Fixed with https://github.com/wso2/identity-apps/pull/2983/commits/a582cef64780cf4b9ffe922d0a8adda99e94f418

File: apps/console/src/features/applications/utils/application-management-utils.ts [hunk 3]
```diff
@@ -399,4 +399,23 @@ export class ApplicationManagementUtils {
 
         return SAMLConfigurationDisplayNames[mode];
     }
+
+    /**
+     * Todo Remove this mapping and fix the backend.
+     */
+     public static mapProtocolTypeToName = ((type: string): string => {
```
Comment (brionmario) [inline apps/console/src/features/applications/utils/application-management-utils.ts:406]:
```suggestion
     public static mapProtocolTypeToName(type: string): string => {
```

File: apps/console/src/features/applications/constants/application-management.ts [hunk 4]
```diff
@@ -449,4 +449,6 @@ export class ApplicationManagementConstants {
      * Error code for invalid metadata URL.
      */
     public static readonly ERROR_CODE_INVALID_METADATA_URL: string = "SAML-60003";
+
+    public static readonly PLAYGROUND_TOUR_STATUS_STORAGE_KEY = "isPlaygroundTourViewed";
```
Comment (induwaram) [inline apps/console/src/features/applications/constants/application-management.ts:453]: @brionmario

Comment (induwaram) [inline apps/console/src/features/applications/constants/application-management.ts:453]: @brionmario

**Comment by induwaram** [comment]
Moved to a new PR - https://github.com/wso2/identity-apps/pull/3042

**Pull request closed without merging**